woocommerce/woocommerce
# Refactor remote specs structure and naming

### Changes proposed in this Pull Request:

Closes #44451.

This PR:

- Deprecates RuleProcessors and Transformers classes using DeprecatedClassFacade to proxy to new classes
- Deprecates DataSourcePoller.php and proxy public functions to new abstract class
- Updates all imports to deprecated classes to new path
- Adds check for DeprecatedClassFacade to ensure no duplicate message in a single request
- Some lint fixes to old files

### How to test the changes in this Pull Request:

#### Regression test 

1. In a fresh site, go to Setup Wizard
2. Click continue
3. Choose `Hong Kong` or other countries that do not support smart shipping defaults
4. Click continue
5. Observe the free features extensions are shown ([screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/5baaaa4c-9042-4266-b03d-f9a78fb37971))
1. Go to WooCommerce > Home
2. Observe inbox notes are displayed, and look for a remote note `Save big on foreign exchange (FX) fees` ([screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/02240367-5c19-41f8-ad9a-ba102182bbc7))
6. Click on `Get paid` task
7. Observe payment list are shown ([screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/e9f07a98-64b2-4a5d-bf52-de6c65911a47))
8. Go back to WooCommerce > Home
9. Click on `Get your products shipped` task
10. Fill in the forms with any value and proceed to step 3
11. Observe shipping list are shown ([screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/f03613a8-17c1-44b2-9ffd-bec47109e31b))
12. Click on `Grow your business` task
13. Observe marketing list are shown ([screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/2b9570c0-4fb0-405f-8798-b15badf62856))
14. Go to `WooCommerce > Settings > Payments`
15. Observe payment extensions are shown ([screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/ec597ec9-8dd8-4df6-ac79-f485c1116f7b))
16. Go to `Payments`
17. Observe there is promo note `Save 10% on payment processing costs in your first three months` ([screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/a5792bcd-a4cd-432b-b92b-d7207c021aed))

#### (Optional) Developer test

##### Test deprecation message
1. After running all above step, look at the debug file, `debug.log`. You **should not** find any deprecated warning such as the following:
`Automattic\WooCommerce\Admin\RemoteInboxNotifications\GetRuleProcessor::get_processor is deprecated since version 8.8.0! Use Automattic\WooCommerce\Admin\RemoteSpecs\RuleProcessors\GetRuleProcessor::get_processor instead.`

##### Test daily job
1. Go to Tools > WCA Test Helper > Tools
2. Click on `Run wc_admin_daily job`
3. Observe it's successful

##### Test specs saved
1. With SQL query tool, run the query `SELECT option_name, option_value FROM wp_options WHERE option_name LIKE '%specs%';`
2. Observe multiple specs are saved and have expiry timestamps that's around 7 days ([screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/6980d0fd-d501-4ad1-95b0-b10b104e8398))




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
